### PR TITLE
Task 5 ticket validation and file parsing functionality

### PR DESCRIPTION
**Pull request by nicatzamanwork**
Task 5 ticket validation and file parsing functionality

### CODE REVIEW

File: src/main/service/TicketService.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    tickets.add(new Ticket(parts[0].trim(), parts[1].trim(), parts[2].trim(), Double.parseDouble(parts[3].trim())));
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return tickets;
+    }
+
+    public void validateTickets(List<Ticket> tickets) {
```
Comment (Darya-Drutsko) [inline src/main/service/TicketService.java:58]: this method is too long, separate it into smaller ones

File: src/main/service/TicketService.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        int totalTickets = tickets.size();
+        int validTickets = 0;
+        Map<String, Integer> violationCounts = new HashMap<>();
+        violationCounts.put("startDate", 0);
```
Comment (Darya-Drutsko) [inline src/main/service/TicketService.java]: create constant for "startDate" and other value here

File: src/main/service/TicketService.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        violationCounts.put("ticketType", 0);
+
+        Date today = new Date();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        for (Ticket ticket : tickets) {
+            boolean isValid = true;
+            List<String> violations = new ArrayList<>();
+
+            if (!List.of("DAY", "WEEK", "MONTH", "YEAR").contains(ticket.getTicketType())) {
```
Comment (Darya-Drutsko) [inline src/main/service/TicketService.java]: "DAY", "WEEK", "MONTH", "YEAR" looks like separate data type :)

**Review by Darya-Drutsko — COMMENTED** [review]
Overall comment: don't forget about formatting and check please tab file changed while creating pr - there should contain only code for the task(not from the previous ones)

File: src/main/service/TicketService.java (hunk 1 above)
Comment (nicatzamanwork) [reply src/main/service/TicketService.java:58]: fixed

File: src/main/service/TicketService.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+public class TicketService {
+
+    private static final String START_DATE = "startDate";
+    private static final String PRICE = "price";
+    private static final String TICKET_TYPE = "ticketType";
+
+    private enum TicketType {
```
Comment (Darya-Drutsko) [inline src/main/service/TicketService.java]: Add new enum to separate file

**Review by Darya-Drutsko — APPROVED** [review]

**Pull request merged**